service/src/main/java/bio/terra/cbas/runsets/monitoring/SmartRunsPoller.java

```diff
@@ -0,0 +1,85 @@
+package bio.terra.cbas.runsets.monitoring;
+
+import static java.util.stream.Collectors.groupingBy;
+
+import bio.terra.cbas.dao.RunDao;
+import bio.terra.cbas.dependencies.wes.CromwellService;
+import bio.terra.cbas.models.Run;
+import cromwell.client.ApiException;
+import cromwell.client.model.State;
+import java.time.OffsetDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+@Component
+public class SmartRunsPoller {
+
+  private final CromwellService cromwellService;
+  private final RunDao runDao;
+
+  private static final org.slf4j.Logger logger = LoggerFactory.getLogger(SmartRunsPoller.class);
+
+  public SmartRunsPoller(CromwellService cromwellService, RunDao runDao) {
+    this.cromwellService = cromwellService;
+    this.runDao = runDao;
+  }
+
+  public List<Run> updateRuns(List<Run> runs) {
```

Comment:
my first reaction here was to want a top-level comment briefly describing what this function does (and perhaps what makes it "smart" polling).

that being said, I basically understand what's happening here, and think it's fine. so I won't let this suggestion block approval

Reply:
I agree. A high level comment would be nice to have.